### PR TITLE
added polar coordinates

### PR DESCRIPTION
**Pull request by fcheadle**
[Polar Coordinates](https://en.wikipedia.org/wiki/Polar_coordinate_system ) are another way of representing  space on a 2d plane. They are measured via a `Radius`, which is the distance from any given `Origin` point, and a `Theta`, which is the degree of rotation around the origin (going clockwise).

I've added Polar Coordinates, translations to and from Cartesian Points, and three example Polar Functions which accept a `Theta` and return a Point.

Those functions are a simple spiral: `return new PolarCoordinate(theta * theta, theta);`, a simple spirograph, and the butterfly curve.

New classes, methods, and properties come with documentation and unit tests.

### CODE REVIEW

File: TheSadRogue.Primitives/PolarCoordinate.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                return new Point((int)Math.Round(x, 0), (int)Math.Round(y, 0));
+            },
+        };
+
+        public Point ToCartesian()
+        {
+            double x = Radius * Math.Cos(Theta);
+            double y = Radius * Math.Sin(Theta);
+            return new Point((int)Math.Round(x, 0), (int)Math.Round(y, 0));
+        }
```
Comment (Chris3606) [inline TheSadRogue.Primitives/PolarCoordinate.cs:113]: Slightly cleaner:
```suggestion
        public Point ToCartesian() => PolarToCartesian(Radius, Theta)
```

File: TheSadRogue.Primitives/PolarCoordinate.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        /// </summary>
+        /// <remarks>
+        /// Each function accepts a parameter of Theta, which is the angle of rotation around the center.
+        /// They return a Point that can be used on the map.
+        /// Polar functions are very slow. Use them seldom, for things like spirographs
+        /// As the radius gets large, there will be increased likelihood
+        /// of functions producing gaps in the output.
+        /// You've been warned.
+        /// </remarks>
+        public static List<Func<double, Point>> PolarFunctions => new List<Func<double, Point>>
```
Comment (Chris3606) [inline TheSadRogue.Primitives/PolarCoordinate.cs:73]: Nice!  Are these just examples, or functions intended for a user to grab out and use?  If they're intended for the user, is there a way to make them more accessible?  Right now they'd have to look at the source and grab an index out.  Perhaps `Dictionary<string, Func<double, Point>>`?

File: TheSadRogue.Primitives/Point.cs [hunk 3]
```diff
@@ -525,6 +525,13 @@ public static (int x, int y) operator /(Point c, (int x, int y) tuple)
         [Pure]
         public bool Equals((int x, int y) other) => X == other.x && Y == other.y;
         #endregion
+
+        #region polar coordinates
+        public PolarCoordinate ToPolarCoordinate()
```
Comment (Chris3606) [inline TheSadRogue.Primitives/Point.cs:530]: Annotation that helps compilers/analyzers for structs, indicates the function does not change members:
```suggestion
        [Pure]
        public PolarCoordinate ToPolarCoordinate()
```

**Comment by fcheadle** [comment]
Added the last bit of feedback! Hope everything is up to standard.

**Pull request merged**